### DISCUSSION TITLE
I'm building a site to be run on an IIS server, are runtime configurations still possible?

### DISCUSSION THREAD

**Discussion by isaackearl** [nuxt/nuxt#19274] [Questions]
I work for a large company building a pretty large app with Vue 3.  Lately I've been working on converting the entire application over to Nuxt 3 so we can make use of Runtime configurations.  Problem is our production servers do not have node (and we can't get it).  We deploy everything through IIS.  

I'm just getting to the point where this conversion to Nuxt is mostly working, but I'm now wanting to make sure that runtime configurations are still possible.  The goal is to be able to drop the files in place but still configure important environment variables such as baseUrl etc.

I understand that I cannot use .env files in production, but I'll still be able to setup ENV variables on my system, and they should be used right?  Or did I misuse my time because I didn't understand that runtime configurations are only available when using a node server... ?

TLDR:
ssr: false
npm run generate
hosting my files through IIS
Can I use environmen variables on my production server to modify runtime configurations?

Thanks!

**Comment by isaackearl** [comment]
Follow up question: if it ends up that there is nothing I can do in this situation.  If I switch to ssr: true and then use npm run generate to bundle it up, can I use env variables with that?  Basically my only limitation is I can't use node to host it, and once I'm deployed, I can't do any builds.  I have to create an artifact and deliver it, and it needs to be configurable.

**Comment by isaackearl** [comment]
I've done some testing and it seems like using npm run generate always leads to a circumstance where you **cannot** use ENV variables at all.  So in effect... Runtime configurations don't work with either SSR or Client side at all as long as you are generating the static site... This should be HIGHLIGHTED in the documentation.  

I've wasted a ton of time trying to make runtime configurations work and they don't for our needs and it's not clear that this is the case in the documentation.
